### PR TITLE
Add `@hash` package with SHA1 implementation

### PR DESCRIPTION
**Pull request by fantix**
Depends on:

- [ ] #329

### CODE REVIEW

File: hash/hash.mbti [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package moonbitlang/core/hash
+
+// Values
+
+// Types and methods
+type SHA1
+fn SHA1::digest(SHA1) -> Bytes
+fn SHA1::make(Bytes) -> SHA1
+fn SHA1::update(SHA1, Bytes) -> Unit
```
Comment (fantix) [inline hash/hash.mbti:9]: These `Bytes` are ideally `@mem.Loadable`, refs moonbitlang/moonbit-docs#190

File: hash/sha1.mbt [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+}
+
+fn update_multi(
+  h : Array[Int],
+  blocks : @bytes.BytesView,
+  n_blocks : Int
+) -> Unit {
+  for i = 0; i < n_blocks; i = i + 1 {
+    let l = (blocks[i * 64..]) as @mem.Loadable
+    let [ ha, hb, hc, hd, he, .. ] = h
```
Comment (fantix) [inline hash/sha1.mbt:48]: This is raising a warning that I don't understand:

```
core/hash/sha1.mbt:48:38-48:39 Warning 011: Partial match, some hints:
[ ]
```

Comment (peter-jerry-ye) [reply hash/sha1.mbt:48]: I assume it means that `h` may be empty

**Comment by bobzhang** [comment]
we would like to have this package landed, could it be submitted as a standalone PR independently

**Comment by fantix** [comment]
> we would like to have this package landed, could it be submitted as a standalone PR independently

Sure, tho I'm outta town for 2 weeks, I can only reply to messages. Feel free to push to my branch, or cherry-pick in whatever way the team prefers, if it's too long to wait for 2 weeks. Thank you and sorry for the inconvenience!

**Comment by bobzhang** [comment]
you are welcome to revive this when you have time, close for now since it is inactive for a while.

**Pull request closed without merging**